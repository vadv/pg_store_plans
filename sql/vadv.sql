SET client_min_messages = 'error';
CREATE EXTENSION IF NOT EXISTS pg_store_plans;
CREATE EXTENSION IF NOT EXISTS pg_stat_statements;
SELECT pg_stat_statements_reset() IS NOT NULL AS t;
SELECT pg_store_plans_reset(); -- collision with plan id
SELECT 'sleep' as regress, pg_sleep(0.1);
SELECT sum(p.calls) FROM pg_stat_statements s JOIN pg_store_plans p ON (s.queryid = p.queryid_stat_statements) JOIN pg_database d ON (d.oid = s.dbid) WHERE  s.query = 'SELECT $1 as regress, pg_sleep($2)';
SET pg_store_plans.min_duration TO '1s';
SELECT 'sleep' as regress, pg_sleep(0.1);
SELECT sum(p.calls) FROM pg_stat_statements s JOIN pg_store_plans p ON (s.queryid = p.queryid_stat_statements) JOIN pg_database d ON (d.oid = s.dbid) WHERE s.query = 'SELECT $1 as regress, pg_sleep($2)';
SET pg_store_plans.slow_statement_duration TO '1ms';
SELECT 'sleep' as regress, pg_sleep(0.1);
SELECT sum(p.calls) FROM pg_stat_statements s JOIN pg_store_plans p ON (s.queryid = p.queryid_stat_statements) JOIN pg_database d ON (d.oid = s.dbid) WHERE s.query = 'SELECT $1 as regress, pg_sleep($2)';
