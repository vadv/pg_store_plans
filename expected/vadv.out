SET client_min_messages = 'error';
CREATE EXTENSION IF NOT EXISTS pg_store_plans;
CREATE EXTENSION IF NOT EXISTS pg_stat_statements;
SELECT pg_stat_statements_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

SELECT pg_store_plans_reset(); -- collision with plan id
 pg_store_plans_reset 
----------------------
 
(1 row)

SELECT 'sleep' as regress, pg_sleep(0.1);
 regress | pg_sleep 
---------+----------
 sleep   | 
(1 row)

SELECT sum(p.calls) FROM pg_stat_statements s JOIN pg_store_plans p ON (s.queryid = p.queryid_stat_statements) JOIN pg_database d ON (d.oid = s.dbid) WHERE  s.query = 'SELECT $1 as regress, pg_sleep($2)';
 sum 
-----
   2
(1 row)

SET pg_store_plans.min_duration TO '1s';
SELECT 'sleep' as regress, pg_sleep(0.1);
 regress | pg_sleep 
---------+----------
 sleep   | 
(1 row)

SELECT sum(p.calls) FROM pg_stat_statements s JOIN pg_store_plans p ON (s.queryid = p.queryid_stat_statements) JOIN pg_database d ON (d.oid = s.dbid) WHERE s.query = 'SELECT $1 as regress, pg_sleep($2)';
 sum 
-----
   2
(1 row)

SET pg_store_plans.slow_statement_duration TO '1ms';
SELECT 'sleep' as regress, pg_sleep(0.1);
 regress | pg_sleep 
---------+----------
 sleep   | 
(1 row)

SELECT sum(p.calls) FROM pg_stat_statements s JOIN pg_store_plans p ON (s.queryid = p.queryid_stat_statements) JOIN pg_database d ON (d.oid = s.dbid) WHERE s.query = 'SELECT $1 as regress, pg_sleep($2)';
 sum 
-----
   3
(1 row)

